### ISSUE TITLE
`trainable` and `requires_grad` have no effect in `BertForClassification` and `BertPooler`

### ISSUE THREAD

**Issue by dcfidalgo** [allenai/allennlp#2927]
**Describe the bug**
The kwarg `trainable` in `BertForClassification` as well as `requires_grad` in `BertPooler` have no effect on freezing/unfreezing the respective parameters in the model.

**To Reproduce**
1. In the file *allennlp/tests/fixtures/bert/bert_pooler.jsonnet* change the `bert_model` to "bert-base-uncased" for example.
2. Run
```
$ allennlp train -s ./bert_bug_test allennlp/tests/fixtures/bert/bert_pooler.jsonnet --include-package allennlp.tests.models -f
```
and look at the
    
    ... Number of trainable parameters: ...

3. Change `requires_grad` in 
```json
        "seq2vec_encoder": {
           "type": "bert_pooler",
           "pretrained_model": bert_model,
           "requires_grad": false
```
to `true` and run the training command again.
4. You can do the same for the `BertForClassification` model with the *allennlp/tests/fixtures/bert/bert_for_classification.jsonnet* 

**Expected behavior**
In the second training run the number of trainable parameters should be higher and the parameters

    _text_field_embedder.token_embedder_bert.bert_model.pooler.dense.weight
    _text_field_embedder.token_embedder_bert.bert_model.pooler.dense.bias

should be listed under `Following parameters are Tunable (with gradient)`

**System (please complete the following information):**
 - OS: OSX, Mojave
 - Python version: 3.7
 - AllenNLP version: master and v0.8.4
 - PyTorch version: 1.1.0

**Additional context**
A fix would be to replace the line 42 in *bert_pooler.py*
```python
    self.pooler.requires_grad = requires_grad
```
with
```python
    for param in self.pooler.parameters():
        param.requires_grad = requires_grad
```
as is also done in the `PretrainedBertEmbedder`.
A similar fix would also work for the `trainable` argument in the `BertForClassification` class.
If you are interested in this fix i would be more than happy to provide a PR.

And by the way, thank you for this awesome package and keep up the amazing work!
Best

**Comment by joelgrus** [comment]
:+1: pr welcome
